### PR TITLE
Issue #11166: remove usages of getFileContents

### PR DESCRIPTION
**Pull request by strkkk**
Part of #11166
Removes usages from
- EmptyLineSeparatorCheck

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java [hunk 1]
```diff
@@ -298,6 +298,15 @@ public final String getLine(int index) {
         return context.get().fileContents.getLine(index);
     }
 
+    /**
+     * Checks if the current file is a package-info.java file.
+     *
+     * @return true if the package file.
+     */
+    public final boolean inPackageInfo() {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java:306]: As to me too much details in abstract class, if we can not avoid checking file name, abstract class should have method to return name for Check to do anything required

**Review by romani — CHANGES_REQUESTED** [review]
Can we split PR in two 
With inPackageInfo
And used lines

?

**Comment by strkkk** [comment]
yes, this is better, since now some pitest issues with appear.

Created https://github.com/checkstyle/checkstyle/pull/11502

File: src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java (hunk 1 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java:306]: changed, see other pr https://github.com/checkstyle/checkstyle/pull/11502

**Comment by romani** [comment]
please update this PR to not have updates for package-info.

**Comment by strkkk** [comment]
> please update this PR to not have updates for package-info.

I will do it right after other PR is merged

**Comment by romani** [comment]
What PR is blocker ?

**Review by pbludov — APPROVED** [review]

**Comment by rnveach** [comment]
@strkkk ping

**Comment by strkkk** [comment]
There are weird pitest failures after my changes, I do not have time to finish it.

**Pull request closed without merging**